types.d.ts

```diff
@@ -1,79 +1,204 @@
-    | 'NEW'
-    | 'PARTIALLY_FILLED'
-    | 'PENDING_CANCEL'
-    | 'REJECTED'
-
-  export type OrderType =
-    | 'LIMIT'
-    | 'LIMIT_MAKER'
-    | 'MARKET'
-    | 'STOP_LOSS'
-    | 'STOP_LOSS_LIMIT'
-    | 'TAKE_PROFIT'
-    | 'TAKE_PROFIT_LIMIT'
-
-  export type TimeInForce = 'GTC' | 'IOC'
+    export default function (options?: { apiKey: string; apiSecret: string }): Binance;
+
+    export interface Account {
+        balances: AssetBalance[];
+        buyerCommission: number;
+        canDeposit: boolean;
+        canTrade: boolean;
+        canWithdraw: boolean;
+        makerCommission: number;
+        sellerCommission: number;
+        takerCommission: number;
+        updateTime: number;
+    }
+
+    export interface AssetBalance {
+        asset: string;
+        free: string;
+        locked: string;
+    }
+
+    export interface Binance {
+        accountInfo(): Promise<Account>;
+
+        order(options: NewOrder): Promise<Order>;
+
+        prices(): Promise<{ [index: string]: string }>;
+
+        time(): Promise<number>;
+
+        ws: WebSocket;
+    }
+
+    export interface WebSocket {
+        depth: (pair: string, callback: (depth: Depth) => void) => Function;
+        partialDepth: (options: { symbol: string, level: number }, callback: (depth: PartialDepth) => void) => Function;
+        ticker: (pair: string, callback: (ticker: Ticker) => void) => Function;
+        allTickers: (callback: (tickers: Ticker[]) => void) => Function;
+        candles: (pair: string, period: string, callback: (ticker: Candle) => void) => Function;
+        trades: (pairs: string[], callback: (trade: Trade) => void) => Function;
+        user: ( callback: (msg: Message) => void) => Function;
+    }
+
+    export interface NewOrder {
+        icebergQty?: string;
+        newClientOrderId?: string;
+        price?: string;
+        quantity: string;
+        recvWindow?: number;
+        side: OrderSide;
+        stopPrice?: string;
+        symbol: string;
+        timeInForce?: TimeInForce;
+        type: OrderType;
+    }
+
+    interface Order {
+        clientOrderId: string;
+        executedQty: string;
+        icebergQty?: string;
+        orderId: number;
+        origQty: string;
+        price: string;
+        side: OrderSide;
+        status: OrderStatus;
+        stopPrice?: string;
+        symbol: string;
+        timeInForce: TimeInForce;
+        transactTime: number;
+        type: OrderType;
+    }
+
+    export type OrderSide = 'BUY' | 'SELL';
+
+    export type OrderStatus =
+        | 'CANCELED'
+        | 'EXPIRED'
+        | 'FILLED'
+        | 'NEW'
+        | 'PARTIALLY_FILLED'
+        | 'PENDING_CANCEL'
+        | 'REJECTED';
+
+    export type OrderType =
+        | 'LIMIT'
+        | 'LIMIT_MAKER'
+        | 'MARKET'
+        | 'STOP_LOSS'
+        | 'STOP_LOSS_LIMIT'
+        | 'TAKE_PROFIT'
+        | 'TAKE_PROFIT_LIMIT';
+
+    export type TimeInForce = 'GTC' | 'IOC';
+}
+
+declare module Results {

```

Comment:
Why do you declare a new module? Can't we put the interfaces inside of that also in the main `'binance-api-node'` module?

Reply:
we totally can, i didnt want to pollute the main module.
would you prefer that?

Reply:
Yipp, I would prefer having it in the main module.